lucene/core/src/java/org/apache/lucene/store/ByteBufferIndexInput.java

```diff
@@ -107,6 +122,24 @@ public final void readBytes(byte[] b, int offset, int len) throws IOException {
     }
   }
 
+  @Override
+  public void readLongs(ByteOrder byteOrder, long[] dst, int offset, int length) throws IOException {
+    try {
+      final int position = curBuf.position();
+      guard.getLongs(curLongBufferViews[position & 0x07].position(position >>> 3), dst, offset, length);
+      if (byteOrder != ByteOrder.nativeOrder()) {
+        for (int i = offset, end = offset + length; i < end; ++i) {
+          dst[i] = Long.reverseBytes(dst[i]);
+        }
+      }
+      curBuf.position(position + (length << 3));
+    } catch (BufferUnderflowException e) {
+      super.readLongs(byteOrder, dst, offset, length);
```

Comment:
Hmm, how come we don't also do the `setCurBuf` here?  It seems like we are falling back to the slow (`super.readLongs`) implementation always on the first `readLongs` call, but since that slow version will use `readByte` it will then initialize the `curLongBufferViews` for subsequent `readLongs` calls?

Reply:
This is correct.